tests/Unit/Controllers/UserStoryControllerTest.php

```
@@ -1030,4 +1147,707 @@ public function scoreReturnsJsonErrorWhenStoryNotFound(): void
+            $this->makePostRequest(['project_id' => '5', 'hl_item_ids' => ['1']]),
+            $this->response,
+            $this->auth,
+            $this->db,
+            $this->config
+        );
+        $ctrl->generate();
+
+        // Gemini fails (no API key in test config) — flash_error set, redirect
+        $this->assertStringContainsString('project_id=5', $this->response->redirectedTo ?? '');
+        $this->assertArrayHasKey('flash_error', $_SESSION);
+    }
```

PHPMD (2.15.0)

[error] generateRedirectsOnGeminiFailureForHlItem accesses the super-global variable $_SESSION. (undefined)

(Superglobals)